src/pybind/mgr/volumes/fs/operations/versions/subvolume_v1.py

```diff
@@ -139,6 +139,12 @@ def remove_clone_source(self, flush=False):
         if flush:
             self.metadata_mgr.flush()
 
+    def add_clone_failure(self, errno, error_msg):
+        self.metadata_mgr.add_section("failure")
```

Comment:
do we want to standardize the format of section header names? I see that section header GLOBAL is all upper case, and so is USER_METADATA.

Maybe name the section header name as CLONE_FAILURE to be explicit?

Not sure about this, but do we need class variables for class MetadataManager similar to GLOBAL_SECTION, GLOBAL_META_KEY_TYPE, etc?
So maybe in class MetadataManager,
  CLONE_FAILURE_SECTION = "CLONE_FAILURE"
  CLONE_FAILURE_META_KEY_ERRNO = "errno"
  CLONE_FAILURE_META_KEY_ERROR_MSG = "error_msg"

Reply:
> do we want to standardize the format of section header names? I see that section header GLOBAL is all upper case, and so is USER_METADATA.

It's not maintained in clone's `.meta` file. The `source` section name is small. So I followed the same.
> 
> Maybe name the section header name as CLONE_FAILURE to be explicit?

The command is explicitly clone status command. So I think that's the reason, it's not used in `source` section as well. I think it is fine.

> 
> Not sure about this, but do we need class variables for class MetadataManager similar to GLOBAL_SECTION, GLOBAL_META_KEY_TYPE, etc? So maybe in class MetadataManager, CLONE_FAILURE_SECTION = "CLONE_FAILURE" CLONE_FAILURE_META_KEY_ERRNO = "errno" CLONE_FAILURE_META_KEY_ERROR_MSG = "error_msg"

Not sure, it's not done for `source` section. Should we do it for all the existing ones as well then ?

Reply:
> > do we want to standardize the format of section header names? I see that section header GLOBAL is all upper case, and so is USER_METADATA.
> 
> It's not maintained in clone's `.meta` file. The `source` section name is small. So I followed the same.
> 

@kotreshhr Where does this get stored?

> > Maybe name the section header name as CLONE_FAILURE to be explicit?
> 
> The command is explicitly clone status command. So I think that's the reason, it's not used in `source` section as well. I think it is fine.
> 
> > Not sure about this, but do we need class variables for class MetadataManager similar to GLOBAL_SECTION, GLOBAL_META_KEY_TYPE, etc? So maybe in class MetadataManager, CLONE_FAILURE_SECTION = "CLONE_FAILURE" CLONE_FAILURE_META_KEY_ERRNO = "errno" CLONE_FAILURE_META_KEY_ERROR_MSG = "error_msg"
> 
> Not sure, it's not done for `source` section. Should we do it for all the existing ones as well then ?

@vshankar what do you think?

Reply:
> It's not maintained in clone's `.meta` file. The `source` section name is small. So I followed the same.
> 
> >

`add_section()` would add a new section in the config which gets written out on flush. Looks like this is adding a section in the metadata config for tracking the failure? (errno/error_msg)?

Reply:
> Not sure about this, but do we need class variables for class MetadataManager similar to GLOBAL_SECTION, GLOBAL_META_KEY_TYPE, etc?
> So maybe in class MetadataManager,
> CLONE_FAILURE_SECTION = "CLONE_FAILURE"
> CLONE_FAILURE_META_KEY_ERRNO = "errno"
> CLONE_FAILURE_META_KEY_ERROR_MSG = "error_msg"

+1

Reply:
Done